components/mpas-ocean/src/shared/mpas_ocn_time_average_coupled.F

```diff
@@ -395,6 +417,39 @@ subroutine ocn_time_average_coupled_accumulate(statePool, forcingPool, timeLevel
 
         end if
 
+        call mpas_pool_get_array(forcingPool, 'avgThermalForcingAtCritDepth', avgThermalForcingAtCritDepth)
+        call mpas_pool_get_subpool(statePool, 'tracers', tracersPool)
+        call mpas_pool_get_array(tracersPool, 'activeTracers', activeTracers, 2)
+        call mpas_pool_get_dimension(tracersPool, 'index_temperature', indexTemperature)
+        call mpas_pool_get_dimension(tracersPool, 'index_salinity', indexSalinity)
+
+
+        ! find vertical level that is just above the critical depth reference level
+        ! this does not account for depression due to ice shelf cavities or sea ice
+        iLevelCritDepth = 1
+        do iLevel = 1, nVertLevels
+           if(refBottomDepth(iLevel) > config_2d_thermal_forcing_depth) then
+              iLevelCritDepth = iLevel-1
```

Comment:
I just wanted to point out that what Matt has here is the same as what we have in the highFrequencyOutput AM. @xylar's logic makes sense to me, so if we decide that this needs to be changed here we need a separate bugfix PR for that AM.

Reply:
@xylar , to follow up on @cbegeman's comment, yes, I copied this code from here: https://github.com/E3SM-Project/E3SM/blob/master/components/mpas-ocean/src/analysis_members/mpas_ocn_high_frequency_output.F#L379-L413
In the module, there are specific comments like:
`! find vertical level that is just above the 100 m reference level`
It is unclear to me why for the high frequency output AM the choice was to use the closest level above the target depth.  I assumed it had to do with not wanting to introduce interpolation and also still obtain information in cases where the next level below the target depth is below the seafloor.  Here is the commit that introduces that code, though it doesn't have much additional context: https://github.com/E3SM-Project/E3SM/commit/25a9279f9265ea24dd95d2ec0f6725d78e8b5839

For this current PR, I can see a few choices about how to handle what layer to use:
1. layer with closest depth shallower than target depth (current)
2. layer with closest depth deeper than target depth
3. layer with closest depth (either direction)
4. interpolation between adjacent layers

Looking over this again, as well as reminding myself how the mask is created to answer your other question, I think option 1 (current implementation) is the best choice, because it is the last layer that is shallower than the refBottomDepth and the only choice that is guaranteed to match the mask.  Let me know if you have other thoughts or I'm not understanding the logic correctly.

Reply:
I still disagree that the current implementation does 1.

`refBottomDepth(i)` is the depth of the bottom of the ith layer. If depth `config_2d_thermal_forcing_depth` was greater than `refBottomDepth(i-1)` but it is less than `refBottomDepth(i), then `config_2d_thermal_forcing_depth` is in layer `i` (not `i-1`).

Reply:
I believe the incorrect calculation in high-frequency output was actually introduced in:
https://github.com/E3SM-Project/E3SM/commit/3c77a2ebfa10b80787bda39ed70ce478df264b6b#diff-904437d562948f083fb74894026c6c469cba2ff0e057a29cc1b33e80534736d9
and copied to the splot you reference.

Reply:
As an example, here are the values of `refBottomDepth` for the 80-layer grid:
```
 refBottomDepth = 1.98678646935581, 4.15411939554175, 6.51847301561518, 
    9.09782457480988, 11.9117370565737, 14.9816089343096, 18.3308398810405, 
    21.9848287330396, 25.9713890184824, 30.3208291170111, 35.0661986371497, 
    40.2435334164226, 45.8920987538932, 52.0547139142045, 58.7780736084832, 
    66.1129821226365, 74.1148329705315, 82.8439984303522, 92.3661273493374, 
    102.752599076403, 114.081046662963, 126.435778783603, 139.908344944848, 
    154.597987883586, 170.612224022118, 188.067210086335, 207.088461553922, 
    227.811080298273, 250.38029644816, 274.951640909234, 301.691130033472, 
    330.775262096312, 362.390624492984, 396.733309415583, 434.007926026355, 
    474.426080833997, 518.204206086992, 565.560782817531, 616.712940125286, 
    671.872076779591, 731.238809447229, 794.997370867892, 863.309316527158, 
    936.307184595767, 1014.08826238255, 1096.70883766392, 1184.17979797662, 
    1276.46376106484, 1373.47419118303, 1475.07667395912, 1581.092497774, 
    1691.30425184573, 1805.46295097985, 1923.29619676545, 2044.51666335532, 
    2168.83040784201, 2295.94444415594, 2425.57314654357, 2557.44346644347, 
    2691.29872299424, 2826.90114580615, 2964.03328758224, 3102.49847295099, 
    3242.1205209147, 3382.74290983668, 3524.2275800194, 3666.4535034604, 
    3809.31514413569, 3952.72088681684, 4096.5915105551, 4240.85873802501, 
    4385.46388801116, 4530.35664731102, 4675.493968649, 4820.83908311944, 
    4966.3606305003, 5112.03189866471, 5257.83015784407, 5403.73608416561, 
    5549.73326321215 ;
```
If you just count the entries, you will see that a depth of 100 m (as in @mark-petersen's original code) would be the 20th index.  This is the layer where `refBottomDepth = 102.752599076403`.  But if you follow the logic of the original code, copied many times, you incorrectly subtract one and get the 19th entry, which is one layer to shallow with `refBottomDepth = 92.3661273493374`.  This is not the layer that contains a depth of 100 m.

Reply:
An important point is the distinction between the depth of the *bottom* of the layer (`refBottomDepth`) and the height (now positive up) of the *middle* of the layer (`refZMid`).  I'm not sure if that's figuring into the confusion (and maybe also for @mark-petersen and @vanroekel before you).  Anyway, I'm still quite sure all these calculations are incorrect.

Reply:
I have made a fix for high-frequency output here:
https://github.com/E3SM-Project/E3SM/pull/6497

Reply:
Thanks for the more detailed explanation, @xylar .  I'll propagate your changes in [#6947](https://github.com/E3SM-Project/E3SM/pull/6497) here

Reply:
Updated in 5c1ac5abc846e2c4d80354e27ab87e38cfcb2a36

Reply:
I agree, the modified code looks good.